File: app/src/main/java/xyz/janficko/sampleapp/ui/splash/SplashActivity.java
package xyz.janficko.sampleapp.ui.splash;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;

import xyz.janficko.sampleapp.commons.Constants;
import xyz.janficko.sampleapp.ui.BaseActivity;
import xyz.janficko.sampleapp.ui.sample.SampleActivity;

public class SplashActivity extends BaseActivity {

    private static final int SPLASH_LOAD_WAIT = 2000;

    @Override
    protected int setLayoutResId() {
        return Constants.NO_LAYOUT;
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        new Handler().postDelayed(
                () -> SampleActivity.start(this),
                SPLASH_LOAD_WAIT
        );
    }
}

File: app/src/main/java/com/margento/sampleapp/ui/splash/SplashActivity.kt
package com.margento.sampleapp.ui.splash

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.margento.sampleapp.common.Constants
import com.margento.sampleapp.ui.main.MainActivity
import kotlinx.coroutines.*
import java.util.concurrent.TimeUnit

class SplashActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        CoroutineScope(Dispatchers.IO).launch {
            delay(TimeUnit.SECONDS.toMillis(Constants.SPLASH_WAIT_SECONDS))
            withContext(Dispatchers.Main) {
                MainActivity.start(this@SplashActivity)
            }
        }
    }
}